pkg/plugin/plugin.go

```diff
@@ -1,21 +1,221 @@
 package plugin
 
 import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"net/http"
 	"os"
+	"path/filepath"
 
+	"github.com/gin-gonic/gin"
 	"github.com/vincenzopalazzo/cln4go/plugin"
+	// "strconv"
 )
 
-type PluginState struct{}
+type PluginState struct {
+	Server   *http.Server
+	Password string
+}
+
+type Login struct {
+	Password string `json:"password" binding:"required"`
+}
+
+func zipSource(source, target string) error {
+	// 1. Create a ZIP file and zip.Writer
+	f, err := os.Create(target)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	writer := zip.NewWriter(f)
+	defer writer.Close()
+
+	// 2. Go through all the files of the source
+	return filepath.Walk(source, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// 3. Create a local file header
+		header, err := zip.FileInfoHeader(info)
+		if err != nil {
+			return err
+		}
+
+		// set compression
+		header.Method = zip.Deflate
+
+		// 4. Set relative path of a file as the header name
+		header.Name, err = filepath.Rel(filepath.Dir(source), path)
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			header.Name += "/"
+		}
+
+		// 5. Create writer for the file header and save content of the file
+		headerWriter, err := writer.CreateHeader(header)
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		f, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		_, err = io.Copy(headerWriter, f)
+		return err
+	})
+}
+
+func copy(src, dst string) (int64, error) {
+	sourceFileStat, err := os.Stat(src)
+	if err != nil {
+		return 0, err
+	}
+
+	if !sourceFileStat.Mode().IsRegular() {
+		return 0, fmt.Errorf("%s is not a regular file", src)
+	}
+
+	source, err := os.Open(src)
+	if err != nil {
+		return 0, err
+	}
+	defer source.Close()
+
+	destination, err := os.Create(dst)
+	if err != nil {
+		return 0, err
+	}
+	defer destination.Close()
+	nBytes, err := io.Copy(destination, source)
+	return nBytes, err
+}
+
+func createDownloadDir() (string, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+	sourcePath := home + "/download_tls"
+	err = os.MkdirAll(sourcePath, 0755)
+	if err != nil {
+		return "", err
+	}
+
+	targetPath := filepath.Join(home, "/.lightning/testnet")
```

Comment:
```suggestion
	targetPath := filepath.Join(home, "/.lightning/testnet")
```

`/.lightning/testnet` is not quite right, for instance on my system the directory is in another place, you can access the dir in the plugin config with the key `lightning-dir`